File: array-string/remove-element/remove-element.go
func removeElement(nums []int, val int) int {
    var k int = 0;
    var i int = 0;

    for (i < len(nums)){
        if (nums[i] != val){
            nums[k] = nums[i];
            k++;
        }
        i++;
    }

    return k;
}

File: array-string/remove-element/remove-element.ts
function removeElement(nums: number[], val: number): number {
    let lastId = nums.length - 1;
    let i = 0;

    while (i < nums.length && i <= lastId)
    {
        if (nums[i] == val)
        {
            nums[i] = nums[lastId];
            lastId--;
            continue;
        }

        i++;
    }

    return i;
};